README.md

```diff
@@ -1,47 +1,77 @@
+- Documentation for people who would like to [contribute to mathlib](https://leanprover-community.github.io/contribute/index.html)
+
+Much of the discussion surrounding mathlib occurs in a
+[Zulip chat room](https://leanprover.zulipchat.com/). Since this
+chatroom is only visible to registered users, we provide an
+[openly accessible archive](https://leanprover-community.github.io/archive/)
+of the public discussions. This is useful for quick reference; for a
+better browsing interface, and to participate in the discussions, we strongly
+suggest joining the chat. Questions from users at all levels of expertise are
+welcomed.
+
+## Transitioning from Lean 3
+
+For users familiar with Lean 3 who want to get up to speed in Lean 4 and migrate their existing
+Lean 3 code we have:
+
+- A [survival guide](https://github.com/leanprover-community/mathlib4/wiki/Lean-4-survival-guide-for-Lean-3-users)
+  for Lean 3 users
+- [Instructions to run `mathport`](https://github.com/leanprover-community/mathport#running-on-a-project-other-than-mathlib)
+  on a project other than mathlib. `mathport` is the tool the community used to port the entirety
+  of `mathlib` from Lean 3 to Lean 4.
 
-* Make sure Lean is not running, and close all instances of VSCode running Lean processes.
-* Get the newest version of `elan`. If you already have installed a version of Lean, you can run
+## Contributing
 
-  ```shell
-  elan self update
-  ```
+The complete documentation for contributing to ``mathlib`` is located
+[on the community guide contribute to mathlib](https://leanprover-community.github.io/contribute/index.html)
 
-  If the above command fails, or if you need to install `elan`, run
+The process is different from other projects where one should not fork the repository.
+Instead write permission for non-master branches should be requested on [Zulip](https://leanprover.zulipchat.com)
+by introducing yourself, providing your GitHub handle and what contribution you are planning on doing.
+You may want to subscribe to the `mathlib4` stream
 
-  ```shell
-  curl https://raw.githubusercontent.com/leanprover/elan/master/elan-init.sh -sSf | sh
-  ```
+### Guidelines
 
-  If this also fails, follow the instructions under `Regular install` [here](https://leanprover-community.github.io/get_started.html).
-* To obtain precompiled `olean` files, run `lake exe cache get`. (Skipping this step means the next step will be very slow.)
-* To build `mathlib4` run `lake build`.
-* To build and run all tests, run `make`.
-* You can use `lake build Mathlib.Import.Path` to build a particular file, e.g. `lake build Mathlib.Algebra.Group.Defs`.
-* If you added a new file, run the following command to update `Mathlib.lean`
+Mathlib has the following guidelines and conventions that must be followed
 
-  ```shell
-  find Mathlib -name "*.lean" | env LC_ALL=C sort | sed 's/\.lean//;s,/,.,g;s/^/import /' > Mathlib.lean
-  ```
```

Comment:
This command should be added to the contributing guidelines, I'd say we should keep it their (in the "contributing" section) for now.

Reply:
good point, I just added that entire bulleted list back under `Contributing`